applications/asset_tracker/src/main.c

```diff
@@ -250,10 +250,13 @@ int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
 	case LWM2M_CARRIER_EVENT_DISCONNECTED:
 		LOG_INF("LWM2M_CARRIER_EVENT_DISCONNECTED");
 		break;
-	case LWM2M_CARRIER_EVENT_READY:
-		LOG_INF("LWM2M_CARRIER_EVENT_READY");
+	case LWM2M_CARRIER_EVENT_LTE_READY:
+		LOG_INF("LWM2M_CARRIER_EVENT_LTE_READY");
 		k_sem_give(&cloud_ready_to_connect);
```

Comment:
If we can avoid the that there will be collision in (D)TLS handshakes happening in the carrier library and the application when this semaphore is given, this is a nice improvement

Reply:
No, there will be handshakes after this event; this only indicates there won't be any more link state changes.
@nordic-hani we need to move this to the `LWM2M_CARRIER_EVENT_REGISTERED` case.